packages/core/src/actors/promise.ts

```diff
@@ -76,9 +76,15 @@ export function fromPromise<T>(
 
       resolvedPromise.then(
         (response) => {
+          if ((self as any)._state.status !== 'active') {
+            return;
+          }
```

Comment:
This is obviously a hack but also... it's the only way, right now, to access your own private state. One could *abuse* `getPersistedState` to do that... but I don't think it's a semantically good idea :p

Alternative approach would be to rely on the dead letter queue but it's not a thing (at least not yet)

Reply:
For context, it avoid this warning:
https://github.com/statelyai/xstate/blob/ed4ed9fc9a72de453dcde60bcd1b17dfd1410be3/packages/core/src/interpreter.ts#L420-L435